File: Section-4-Injections/example1_secure_code.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.ResultSet;

public class CustomerProfile {
    public String getCustomerProfileById(String customerId) {
        String query = "SELECT * FROM Customers WHERE CustomerId = ?";

        try (Connection connection = DriverManager.getConnection(connectionString, "username", "password");
             PreparedStatement statement = connection.prepareStatement(query)) {

            statement.setString(1, customerId);
            ResultSet resultSet = statement.executeQuery();

            if (resultSet.next()) {
                return "Name: " + resultSet.getString("Name") + ", Email: " + resultSet.getString("Email");
            } else {
                return "Customer not found.";
            }
        } catch (Exception e) {
            e.printStackTrace();
            return "Error retrieving customer profile.";
        }
    }

    public static void main(String[] args) {
        CustomerProfile profile = new CustomerProfile();
        System.out.println(profile.getCustomerProfileById("123"));
    }
}

File: Section-4-Injections/example1_secure_code.py
import sqlite3

class SocialMedia:
    def get_profile_by_id(self, user_id):
        connection = sqlite3.connect("social_media.db")
        cursor = connection.cursor()

        # Secure against SQL Injection
        query = "SELECT * FROM users WHERE user_id = ?"
        cursor.execute(query, (user_id,))
        result = cursor.fetchone()

        connection.close()

        if result:
            return f"Name: {result[1]}, Email: {result[2]}"
        else:
            return "User not found."

# Example usage
social_media = SocialMedia()
print(social_media.get_profile_by_id("1"))

